contracts/strategy/liquity/LiquityStrategy.sol

```diff
@@ -49,12 +50,15 @@ contract LiquityStrategy is
     bytes32 public constant MANAGER_ROLE = keccak256("MANAGER_ROLE");
     // role for calling harvest() and reinvestRewards()
     bytes32 public constant KEEPER_ROLE = keccak256("KEEPER_ROLE");
+    // role for managing swap targets whitelist
+    bytes32 public constant SETTINGS_ROLE = keccak256("SETTINGS_ROLE");
```

Comment:
Why do we need a SETTINGS_ROLE? Isn't the DEFAULT_ADMIN_ROLE enough for this?

Reply:
This was my dilemma also, that is why I assigned the SETTINGS role to the admin. I made the decision to add the new role because that's how we handled setting the slippage `maxLossOnWithdraw` in the yearn strategy, this felt like a similar situation. This will enable us to assign a separate account apart from admin for controlling the whitelist if we want to, but maybe that's not worth the added complexity. Not really sure what's the best approach tbh

Reply:
I dont think 0xapi will upgrade their core contract more than twice a year.

Reply:
@dxganta I'm trying to understand if that's a gut feeling or if you got that information from someone. We can always ask them, but the idea here is that we need to understand the process of updating this whitelist before we can make this decision. Who's going to do it and etc.

Reply:
I did not ask anyone but its not totally a gut feeling either. I couldn't quite figure out the contract creation date because it was not clear on etherscan but the looking at the analytics https://etherscan.io/address/0xdef1c0ded9bec7f1a1670819833240f027b25eff#analytics it was created around Jun 2020 and started gaining decent traffic since August 2021, which is 1 year from now. So that says they have not changed the address in the last 1 year. Moreover the contract is proxy, so even if they do want to change the contract logic, the address will most probably remain same unless they want a completely new version. So I would say its safe to assume that the address wont change more than twice a year.

Reply:
I'm sorry man, but that's just a gut feeling. Past behavior does not predict the future because they are also just people that may have/want to change how they operate. They may change the architecture to rely on multiple contracts instead of one, or they may want to build features that can't be upgraded, and so much stuff can happen. It's not like they commit to not changing the address. But that doesn't matter for this discussion, I think. To me, it's still not clear who will "review" the new contract when it breaks and who has to approve a change to the allowlist. If it's the admin, we might as well remove the new role, although to make it more flexible in the future, we may want to keep it.